plugin_ide.ui/src/com/github/rustdt/ide/ui/editor/RustEditorActionContributor.java

```diff
@@ -28,6 +47,88 @@ protected RustOpenDefinitionOperation createOpenDefinitionOperation(ITextEditor
 	
 	@Override
 	protected void registerOtherEditorHandlers() {
+		setupOpenType();

```

Comment:
Yeah this is wrong. First, the code to manage the index should be in the core plugin, not UI. Second it needs to be initialized as one of the plugin singletons, on plugin startup. Like RustBundleModelManager for example.

As this will be closely related to SourceModelManager, it might be best to add the "IndexManager" as component of SourceModelManager.